daemon/gluster-blockd.c

```diff
@@ -508,6 +510,35 @@ blockNodeSanityCheck(void)
   if (GB_ASPRINTF(&global_opts, GB_TGCLI_GLOBALS, gbConf->configShellLogFile) == -1) {
     return ENOMEM;
   }
+
+  tmp = gbRunnerGetOutput(TARGETCLI_VERSION);
+  if (!gbDependencyVersionCompare(TARGETCLI_DAEMON, tmp)) {
+    use_targetclid = false;
+    LOG("mgmt", GB_LOG_WARNING,
+        "targetclid required: %s, current version: %s, using cli mode",
+        GB_MIN_TARGETCLI_DAEMON_VERSION, tmp);
+  }
+  GB_FREE(tmp);
+
+  if (use_targetclid) {
+    tmp = global_opts;
+    /* Check if targetclid is running */
+    ret = gbRunner("ps aux ww | grep -w '[t]argetclid' > /dev/null");
+    if (ret) {
+      LOG("mgmt", GB_LOG_WARNING, "targetclid not running, using targetcli");
+      if (GB_ASPRINTF(&global_opts, "targetcli --skip-daemon; %s", tmp) == -1) {
```

Comment:
BTW, should we also set the auto_use_daemon=false at the same time here since the targetclid is not running ?

Reply:
@lxbsz '# targetcli --skip-daemon' will internally set auto_use_daemon=false, as there is no daemon running, we cannot send any commands via daemon, hence ''# targetcli --skip-daemon' will internally bypass call to daemon as it is not running and set 'auto_use_daemon=false' via the cli route.

Reply:
@pkalever 
Ah, yes, you are right. I misread that part of code.
Thanks.

Reply:
@lxbsz could you please test this and approve ? Thanks!

Reply:
@pkalever 
Before I had test it and it works. I thought we need to wait the targetcli_fb patches, which were not totally finished yet and the '--skip-daemon' option may change ?

Reply:
@lxbsz 
That is right, we need to wait for targetcli pathes for sure, just wanted to make sure we keep this patch ready till then.

Reply:
@pkalever Sure, once that PR is done I will test it again and approve it.

Reply:
@lxbsz the targetcli improvements PR is merged now, could you please test and review this patch. Thanks!